bitnami/kafka/values-production.yaml

```diff
@@ -363,6 +364,32 @@ metrics:
 #        cpu: 100m
 #        memory: 100Mi
 
+    service:
+      ## Kafka Exporter Service type
+      ##
+      type: ClusterIP
+      ## Kafka Exporter Prometheus port
+      ##
+      port: 9308
+      ## Specify the nodePort value for the LoadBalancer and NodePort service types.
+      ## ref: https://kubernetes.io/docs/concepts/services-networking/service/#type-nodeport
+      ##
+      nodePort: ""
+      ## Set the LoadBalancer service type to internal only.
+      ## ref: https://kubernetes.io/docs/concepts/services-networking/service/#internal-load-balancer
+      ##
+      # loadBalancerIP:
+      ## Set the Cluster IP to use
+      ## ref: https://kubernetes.io/docs/concepts/services-networking/service/#choosing-your-own-ip-address
+      ##
+      # clusterIP: None
+      ## Annotations for the Kafka Exporter Prometheus metrics service
+      ##
+      annotations:
+        prometheus.io/scrape: "true"
+        prometheus.io/port: "{{ .Values.metrics.kafka.port }}"
```

Comment:
can you really use helm templates in the values.yaml file?

Reply:
Yes, thanks to the macro below:
```
{{/*
Renders a value that contains template.
Usage:
{{ include "kafka.tplValue" ( dict "value" .Values.path.to.the.Value "context" $) }}
*/}}
{{- define "kafka.tplValue" -}}
    {{- if typeIs "string" .value }}
        {{- tpl .value .context }}
    {{- else }}
        {{- tpl (.value | toYaml) .context }}
    {{- end }}
{{- end -}}
```

You can use:

```
  annotations: {{ include "kafka.tplValue" ( dict "value" .Values.metrics.jmx.service.annotations "context" $) | nindent 4 }}
```

And it will: 
a) render the values
b) use them as a yaml value

Reply:
ah! got it.

Reply:
@sameersbn  @juan131  just to mention that this is breaking the kafka at the moment for me. if we don't specify any port for exporter , prometheus starts to scrape from 9092 which is going to pull down the whole kafka in kubernetes. I ended up having this error/warning all over and no message got received after a while. This hit us badly and our customer was impacted. :( 

`[2020-06-21 07:29:21,062] WARN [SocketServer brokerId=1] Unexpected error from /10.244.5.204; closing connection (org.apache.kafka.common.network.Selector)
org.apache.kafka.common.network.InvalidReceiveException: Invalid receive (size = 1195725856 larger than 104857600)
at org.apache.kafka.common.network.NetworkReceive.readFrom(NetworkReceive.java:105)
	at org.apache.kafka.common.network.KafkaChannel.receive(KafkaChannel.java:448)
	at org.apache.kafka.common.network.KafkaChannel.read(KafkaChannel.java:398)
	at org.apache.kafka.common.network.Selector.attemptRead(Selector.java:678)
	at org.apache.kafka.common.network.Selector.pollSelectionKeys(Selector.java:580)
	at org.apache.kafka.common.network.Selector.poll(Selector.java:485)
	at kafka.network.Processor.poll(SocketServer.scala:861)
	at kafka.network.Processor.run(SocketServer.scala:760)
	at java.base/java.lang.Thread.run(Thread.java:834)
`
so in my opinion scrape should happen via service port `9308` as a default in helm chart.

Reply:
Hi @shapeofarchitect,
    This was merge some time ago, do you mind to open a new issue so we can do a proper handling of it ? This way it will be easier to keep a conversation, add new PRs, etc.

Thanks forehand.

Reply:
yes I will open up a new issue.

Reply:
Thanks !